From bc392daeeaafc27b342349f44db8bbd0a702fee4 Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Tue, 12 Jun 2012 16:31:42 +0900
Subject: [PATCH 37/80] rcarvin: enable the ARGB1555 format conversion

This enables the ARGB1555 format conversion by VIN.
Note: the alpha bit must be fixed with zero.

Signed-off-by: Katsuya Matsubara <matsu@igel.co.jp>
---
 drivers/media/video/rcarvin.c |   14 +++++++++++++-
 1 files changed, 13 insertions(+), 1 deletions(-)

diff --git a/drivers/media/video/rcarvin.c b/drivers/media/video/rcarvin.c
index ba48448..81bc67d 100644
--- a/drivers/media/video/rcarvin.c
+++ b/drivers/media/video/rcarvin.c
@@ -362,7 +362,8 @@ static void free_buffer(struct videobuf_queue *vq,
 /* VnDMR */
 #define VIN_VNDMR_EVA        (0x00010000)  /* EVen field Address offset */
 #define VIN_VNDMR_BPSM       (0x00000010)  /* Byte Position Swap Mode */
-#define VIN_VNDMR_DTMD_YCSEP (0x00000002)  /* transfer: YC separate */
+#define VIN_VNDMR_DTMD_YCSEP    (0x00000002)  /* transfer: YC separate */
+#define VIN_VNDMR_DTMD_ARGB1555 (0x00000001)  /* transfer: ARGB1555 */
 
 /* VnDMR2 */
 #define VIN_VNDMR2_FPS             (0x80000000)  /* Field Polarity Select */
@@ -452,6 +453,10 @@ static int rcar_vin_capture(struct rcar_vin_dev *pcdev)
 		dmr = 0;
 		mc = field | VIN_VNMC_VUP | VIN_VNMC_BPS;
 		break;
+	case V4L2_PIX_FMT_RGB555X:
+		dmr = VIN_VNDMR_DTMD_ARGB1555;
+		mc = field | VIN_VNMC_VUP;
+		break;
 	case V4L2_PIX_FMT_RGB565:
 		dmr = 0;
 		mc = field | VIN_VNMC_VUP;
@@ -1029,6 +1034,13 @@ static const struct soc_mbus_pixelfmt rcar_vin_formats[] = {
 		.packing		= SOC_MBUS_PACKING_NONE,
 		.order			= SOC_MBUS_ORDER_LE,
 	},
+	{
+		.fourcc			= V4L2_PIX_FMT_RGB555X,
+		.name			= "ARGB1555",
+		.bits_per_sample	= 16,
+		.packing		= SOC_MBUS_PACKING_NONE,
+		.order			= SOC_MBUS_ORDER_LE,
+	},
 };
 
 /* This will be corrected as we get more formats */
-- 
1.7.0.4

